### ISSUE TITLE
get method is not write proof

### ISSUE THREAD

**Issue by ghost** [kswap/LRU-Cache#2]

**Comment by bpsingh123** [comment]
Yes, you are trying to modify the collections using read lock which is unsafe

**Comment by aniket-somwanshi** [comment]
Quick tip, If you are modifying a shared variable(writing) use the writeLock().
ReadLock() will allow multiple threads to read at the same time. So if we're writing mid-way, then it can give wierd results. Good implementation otherwise mate!
